examples/asrc_demo/src/rate_server.c

```diff
@@ -423,3 +404,54 @@ uint32_t sum_array(uint32_t * array_to_sum, uint32_t array_length)
     }
     return acc;
 }
+
+typedef struct
+{
+    uint64_t mant;
+    int32_t exp;
+}float_u64_t;
+
+static float_u64_t float_div_u64(float_s32_t dividend, float_s32_t divisor)
```

Comment:
would be nice to see unit tests for this fancy maths. Won't gate PR on it though. 

Reviewing functions like this (fancy maths) by hand won't be nearly as effective use of time as writing unit tests

Reply:
Yes, I am going to write unit tests for these. Will create a separate issue

Reply:
I've created an issue for this. https://xmosjira.atlassian.net/browse/AP-167